pages/api/_middleware.ts

```diff
@@ -1,23 +1,45 @@
-import type { NextApiRequest } from 'next'
-import { getToken } from 'next-auth/jwt'
-import { NextResponse } from 'next/server'
+import { IncomingHttpHeaders } from "http2";
+import type { NextApiRequest } from "next";
+import { getToken } from "next-auth/jwt";
+import { NextResponse } from "next/server";
+
+interface CustomHeader {
+  get: (key: string) => string
+}
+
+type CombinedHeader = IncomingHttpHeaders & CustomHeader;
 
 const testnet = process.env.NEXT_E2E_TESTNET === "true";
 
 const blackList = {
-  GET: ['auth', 'past-events', 'seo', 'users', 'graphql'],
-  POST: ['issue']
-}
-
+  POST: ["network", "pull-request", "repos"],
```

Comment:
I'm thinking this could be reverted to how it was:

We have a list of things that _don't need_ a token and if the url is not inside then we check if a token exists. Otherwise this list will ... grow. like: gigantic. It's easier to mantaign if we just list what we allow without token